### DISCUSSION TITLE
VectorCAN---CANFD

### DISCUSSION THREAD

**Discussion by yoimiya52** [collin80/SavvyCAN#753] [Q&A]
Hi, I would like to know how SavvyCan interacts with VectorCAN and receives canfd messages; I have implemented receiving can messages. Thanks a lot.

**Comment by collin80** [comment]
The honest answer is, I don't know. I don't have any Vector hardware to test with. CAN-FD has been partially implemented in SavvyCAN and so it should work. It ought to be possible to capture FD traffic and see it show up on the main screen. After that, things are a little more hazy. I think loading and saving should work in formats that support CAN-FD. But, a lot of the reverse engineering screens in SC don't fully support CAN-FD yet.

**Reply by yoimiya52** [reply]
Thank you very much for your reply, I refactored the code about Vector connection using Vector's driver; unfortunately, I'm not familiar with the use of Qt, and example  when dealing with QMetaObject::invokeMethod(this, "stop", Qt::BlockingQueuedConnection), the program crashed directly! I can't find the reason. I am very much looking forward to some help from you.

**Reply by yoimiya52** [reply]
I probably know the reason why QCanBus can't capture canfd messages in VectorCAN, it fails in setting the baud rate, bitrate, and canfd_key, I didn't find the reason so I ditched QCanBus.

**Reply by yoimiya52** [reply]
I've solved the problem and I'm sorry for the interruption. In my Vector connection, I was using a timer to receive messages, and there was a ‘while(True)’ inside the timer that caused the event loop for this object to block and not be able to process other events, so the program crashed.
